docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+---
+title: Migrating from MvvmLight
+author: jamesmcroft
+description: This article describes how to migrate MvvmLight solutions to the Windows Community Toolkit MVVM framework.
+keywords: windows 10, uwp, windows community toolkit, uwp community toolkit, uwp toolkit, mvvm, mvvmlight, net core, net standard
+dev_langs:
+  - csharp
+---
+
+# Migrating from MvvmLight
+
+This article outlines some of the key differences between the [MvvmLight Toolkit](https://github.com/lbugnion/mvvmlight) and the MVVM Toolkit to ease your migration. 
+
+While this article specifically focuses on the migrations from MvvmLight to the MVVM Toolkit, note that there are additional improvements that have been made within the MVVM Toolkit so I highly recommend taking a look at the documentation for the APIs. 
+
+## Installing the WCT MVVM Toolkit
+
+To take advantage of the Windows Community Toolkit MVVM framework, you'll first need to install the latest NuGet package to your existing Windows application.
+
+### Install via .NET CLI
+
+```
+dotnet add package Microsoft.Toolkit.Mvvm --version x.x.x
+```
+
+### Install via PackageReference
+
+```
+<PackageReference Include="Microsoft.Toolkit.Mvvm" Version="x.x.x" />
+```
+
+## Migrating ObservableObject
+
+The following steps focus on migrating your existing components which take advantage of the `ObservableObject` of the MvvmLight Toolkit. 
+
+The Windows Community Toolkit MVVM framework provides an [`ObservableObject`](ObservableObject) type that is similar. 
+
+The first change here will be swapping using directives in your components.
+
+```csharp
+// MvvmLight
+using GalaSoft.MvvmLight;
+
+// Toolkit.Mvvm
+using Microsoft.Toolkit.Mvvm.ComponentModel;
+```
+
+Below are a list of migrations that will need to be performed if being used in your current solution.
+
+### ObservableObject Methods
+
+#### Set<T> ( Expression, ref T, T )
+
+`Set(Expression, ref T, T)` does not have a like-for-like method signature replacement. 
+
+However, `SetProperty(ref T, T, string)` provides the same functionality with additional performance benefits. 
+
+```csharp
+// MvvmLight
+this.Set(() => this.MyProperty, ref this.myProperty, value);
+
+// Toolkit.Mvvm
+this.SetProperty(ref this.myProperty, value, nameof(this.MyProperty));
```

Comment:
Just a note, is the usage of `this` just a stylistic choice?
I just wonder whether it might throw off less experience devs, thinking it might be necessary? 🤔